src/PurlServiceProvider.php

```diff
@@ -27,12 +25,18 @@ public function alter(ContainerBuilder $container)
     $assemblerDefinition->addArgument(new Reference('purl.context_helper'));
     $assemblerDefinition->addArgument(new Reference('purl.matched_modifiers'));
 
-    if ($routerDefinition = $container->getDefinition('router.route_provider')) {
-      if ($routerDefinition->getClass() == 'Drupal\Core\Routing\RouteProvider') { // don't break tests
-        $routerDefinition->setClass(PurlRouteProvider::class);
-        $routerDefinition->addArgument(new Reference('purl.context_helper'));
-        $routerDefinition->addArgument(new Reference('purl.matched_modifiers'));
-      }
-    }
+    $routerDefinition = $container->getDefinition('router.route_provider');
```

Comment:
What does this do outside of a testing environment, when all the core services are loaded in and ready to go? It looks like it'll break regular, non-testing installs by appending a bunch of junk arguments to the constructor call.